Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalCheckoutRequest.h

```diff
@@ -0,0 +1,70 @@
+#if __has_include(<Braintree/BraintreePayPal.h>)
+#import <Braintree/BTPayPalRequest.h>
+#else
+#import <BraintreePayPal/BTPayPalRequest.h>
+#endif
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ Payment intent.
+
+ @note Must be set to sale for immediate payment, authorize to authorize a payment for capture later, or order to create an order. Defaults to authorize. Only works in the checkout (one-time payment) flow.
+
+ @see https://developer.paypal.com/docs/integration/direct/payments/capture-payment/ Capture payments later
+ @see https://developer.paypal.com/docs/integration/direct/payments/create-process-order/ Create and process orders
+*/
+typedef NS_ENUM(NSInteger, BTPayPalRequestIntent) {
+    /// Authorize
+    BTPayPalRequestIntentAuthorize = 1,
+
+    /// Sale
+    BTPayPalRequestIntentSale,
+
+    /// Order
+    BTPayPalRequestIntentOrder,
+};
+
+@interface BTPayPalCheckoutRequest : BTPayPalRequest
+
++ (instancetype)new __attribute__((unavailable("Please use initWithAmount:")));
+
+/**
+ Base initializer - do not use.
+ */
+- (instancetype)init __attribute__((unavailable("Please use initWithAmount:")));
+
+/**
+ Initialize a PayPal request with an amount for a one-time payment.
+
+ @param amount Used for a one-time payment. Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.', optional thousands separator ',', and is limited to 7 digits before the decimal point.
+ @return A PayPal request.
+*/
+- (instancetype)initWithAmount:(NSString *)amount;
+
+/**
+ Used for a one-time payment.
+
+ Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.', optional thousands separator ',', and is limited to 7 digits before the decimal point.
+*/
+@property (nonatomic, readonly, strong) NSString *amount;
+
+/**
+ Optional: A valid ISO currency code to use for the transaction. Defaults to merchant currency code if not set.
```

Comment:
do we know which ISO standard this follows? Is it 2 or three characters? We should include this detail with an example

Reply:
Looks like we have more info in the [Braintree docs](https://developers.braintreepayments.com/guides/paypal/checkout-with-paypal/ios/v4#currency-presentment), which link out to the [PayPal docs](https://developer.paypal.com/docs/api/reference/currency-codes/). I can include the PayPal docs link here and specify that it's three-character ISO-4217.